jobs/pacman-awsrules/src/main/java/com/tmobile/cloud/azurerules/VirtualMachine/EncryptionAppTierRule.java

```diff
@@ -98,8 +98,15 @@ private boolean checkIsAppTierAzureVmDiskEncryptionEnabled(String esUrl, Map<Str
                 JsonObject jsonDataItem = (JsonObject) ((JsonObject) hitsJsonArray.get(0))
                         .get(PacmanRuleConstants.SOURCE);
                 logger.debug("Validating the data item: {}", jsonDataItem.toString());
-                JsonArray diskJsonArray = jsonDataItem.getAsJsonObject()
-                        .get(PacmanRuleConstants.DISKS).getAsJsonArray();
+                JsonArray diskJsonArray = null;
+
+                if (jsonDataItem != null && jsonDataItem.isJsonObject()) {
+                    JsonObject jsonObject = jsonDataItem.getAsJsonObject();
+
+                    if (jsonObject.has(PacmanRuleConstants.DISKS) && jsonObject.get(PacmanRuleConstants.DISKS).isJsonArray()) {
+                        diskJsonArray = jsonObject.get(PacmanRuleConstants.DISKS).getAsJsonArray();
+                    }
+                }
                 if(diskJsonArray.size()>0) {
```

Comment:
diskJsonArray can throw NPE on 110 if any of the above conditions are false

Reply:
addressed in latest push